sender_pool.go

```diff
@@ -119,25 +130,39 @@ func WithMaxSenders(count int) LineSenderPoolOption {
 	}
 }
 
-// Acquire obtains a LineSender from the pool. If the pool is empty, a new
+// Sender obtains a LineSender from the pool. If the pool is empty, a new
 // LineSender will be instantiated using the pool's config string.
-func (p *LineSenderPool) Acquire(ctx context.Context) (LineSender, error) {
+// If there is already maximum number of senders obtained from the pool,
+// this calls will block until one of the senders is returned back to
+// the pool by calling sender.Close().
+func (p *LineSenderPool) Sender(ctx context.Context) (LineSender, error) {
+	var (
+		s   LineSender
+		err error
+	)
+
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
 	if p.closed {
-		return nil, fmt.Errorf("cannot Acquire a LineSender from a closed LineSenderPool")
+		return nil, errors.New("cannot Acquire a LineSender from a closed LineSenderPool")
 	}
 
-	if len(p.senders) > 0 {
+	// We may have to wait for a free sender
+	for len(p.freeSenders) == 0 && p.numSenders == p.maxSenders {
+		p.cond.Wait()
+	}
+
```

Comment:
We should return an error here if the pool is closed

Reply:
Nice catch. Added the check here: aa2e8896186240c3b98f3935299c28f1c104e267